community/kernel/src/main/java/org/neo4j/graphdb/Result.java

```diff
@@ -227,15 +227,17 @@
     /**
      * This is the visitor interface you need to implement to use the {@link Result#accept(ResultVisitor)} method.
      */
-    interface ResultVisitor
+    interface ResultVisitor<VisitationException extends Exception>
```

Comment:
I'd prefer not. It would be nice if we could treat the `org.neo4j.helpers` package as internal and not expose it in the API.